securedrop/journalist_templates/admin.html

```diff
@@ -38,4 +38,54 @@
 <p>No users to display</p>
 {% endif %}
 
+<h2>Add/Delete Label Types</h2>
+<p><b>Operational Security Warning:</b> Note that these labels are stored in plaintext on the
+SecureDrop server.</p>
+
+<p>
+  <form id="source_tag_creation" method="post">
+    <input name="csrf_token" type="hidden" value="{{ csrf_token() }}" />
+    <input name="text_to_display" type="text" placeholder="Label Text" />
+    <button type="submit" class="btn" formaction="/admin/create_source_label_type/{{ text_to_display }}"><i class="fa fa-plus"></i> Add source label</button>
+    {% with errors = get_flashed_messages(category_filter=["source"]) %}{% if errors %}
```

Comment:
Nice! It's important to disambiguate handling this set of flashed messages from the set that is handled in `journalist_templates/flashed.html`, which is included in `journalist_templates/base.html`. Using specific categories is a good way to do this, but I would encourage you to use a more explicit name than `source`.

We reviewed `journalist_templates/flashed.html` and determined that it will only flash messages with category `notification` or `error`, so you are safe from the templates accidentally flashing the same message twice.